src/async-queue.js

```diff
@@ -0,0 +1,25 @@
+(function (global) {
+  'use strict';
+
+  const AsyncQueue = () => {
+    let ends = Promise.defer();
+    return {
+      put(value) {
+        const next = Promise.defer();
+        ends.resolve({
+          head: value,
+          tail: next.promise,
+        })
+        ends.resolve = next.resolve;
+        ends.reject = next.reject;
```

Comment:
Is ends.reject used anywhere?

Reply:
I don't get it. 
   * I don't see how the ends object could escape AsyncQueue. It seems safely encapsulated.
   * I looked, and forEach uses the .reject of a deferred, which makes sense. AFAICT, it does not use the .reject of an ends object.